samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/servlet/event/ExportEventLogServlet.java

```diff
@@ -0,0 +1,128 @@
+import org.sakaiproject.samigo.util.SamigoConstants;
+import org.sakaiproject.time.api.UserTimeService;
+import org.sakaiproject.tool.assessment.data.dao.assessment.EventLogData;
+import org.sakaiproject.tool.assessment.services.assessment.EventLogService;
+import org.sakaiproject.tool.assessment.ui.servlet.SamigoBaseServlet;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ExportEventLogServlet extends SamigoBaseServlet {
+
+
+    private static final String CSV_SEPARATOR = ",";
+    private static final DateTimeFormatter EXPORT_DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+    private static final ResourceBundle resourceBundle = ResourceBundle.getBundle(SamigoConstants.EVENT_LOG_BUNDLE);
+
+    private UserTimeService userTimeService = ComponentManager.get(UserTimeService.class);
+
+    public static final String PARAM_SITE_ID = "siteId";
+    public static final String PARAM_ASSESSMENT_ID = "assessmentId";
+
+
+    public void doGet(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
+        // Required
+        String siteId = StringUtils.trimToNull(req.getParameter(PARAM_SITE_ID));
+        // Optional
+        String assessmentId = req.getParameter(PARAM_ASSESSMENT_ID);
+
+        if (siteId == null) {
+            res.sendError(HttpServletResponse.SC_BAD_REQUEST, "Required param siteId is not present.");
+            return;
+        }
+
+        if (!canExportEventLog(siteId)) {
+            Optional<String> optUserId = getUserId();
+            if (optUserId.isPresent()) {
+                log.warn("User with id [{}] is not allowed to export event log for site [{}]", optUserId.get(), siteId);
+            } else {
+                log.warn("Unauthenticated user tried to export event log for site [{}]", siteId);
+            }
+
+            res.sendError(HttpServletResponse.SC_FORBIDDEN);
+            return;
+        }
+
+        if ((assessmentId != null && !NumberUtils.isParsable(assessmentId))) {
+            res.sendError(HttpServletResponse.SC_BAD_REQUEST, "Value of optional param assessmentId is invalid.");
+            return;
+        }
+
+        String filename = StringUtils.replace(resourceBundle.getString("log"), " ", "_");
+        res.setContentType("text/csv");
+        res.setHeader("Content-Disposition", "\"attachment;filename=\"" + filename + ".csv\";");
+
+        EventLogService eventLogService = new EventLogService();
+        List<EventLogData> eventLogDataList;
+        if (assessmentId != null) {
+            eventLogDataList = eventLogService.getEventLogData(siteId, Long.valueOf(assessmentId));
+        } else {
+            eventLogDataList = eventLogService.getEventLogDataBySiteId(siteId);
+        }
+
+        CsvEscaper csvEscaper = new CsvEscaper();
+        PrintWriter out = res.getWriter();
+
+        // Write headers
+        out.print(resourceBundle.getString("title") + CSV_SEPARATOR);
```

Comment:
I feel we must include all the values inside double quotes, I'm tempted to use a library like OpenCSV to prevent issues with Excel.

WDYT?